packages/host/tests/integration/store-test.gts

```diff
@@ -875,4 +880,246 @@ module('Integration | Store', function (hooks) {
+    assert.strictEqual(
+      store.getReferenceCount(jade),
+      1,
+      `reference count for ${jade} is 1`,
+    );
+    assert.strictEqual(
+      store.getReferenceCount(hassan),
+      0,
+      `reference count for ${hassan} is 0`,
+    );
+
+    driver.showComponent = false;
+    await waitFor(`[data-test-rendered-card]`, { count: 0 });
+    assert.strictEqual(
+      store.getReferenceCount(jade),
+      0,
+      `reference count for ${jade} is 0`,
+    );
+    assert.strictEqual(
+      store.getReferenceCount(hassan),
+      0,
+      `reference count for ${hassan} is 0`,
+    );
+  });
+
+  test('reference count is balanced when used with SearchResource that is destroyed', async function (assert) {
+    class Driver {
+      @tracked showComponent = false;
+      @tracked id: string | undefined;
+    }
+
+    let driver = new Driver();
+
+    class ResourceConsumer extends GlimmerComponent {
+      resource = getSearch(this, () =>
+        driver.id
+          ? {
+              filter: {
+                on: baseCardRef,
+                eq: {
+                  id: driver.id,
+                },
+              },
+            }
+          : undefined,
+      );
+      get card() {
+        return this.resource.instances[0];
+      }
+      get renderedCard() {
+        return this.card?.constructor.getComponent(this.card);
+      }
+      <template>
+        {{#if this.card}}
+          <this.renderedCard data-test-rendered-card={{this.card.id}} />
+        {{/if}}
+      </template>
+    }
+
+    await renderComponent(
+      class TestDriver extends GlimmerComponent {
+        <template>
+          {{#if driver.showComponent}}
+            <ResourceConsumer />
+          {{/if}}
+          <CardPrerender />
+        </template>
+      },
+    );
+
+    driver.showComponent = true;
+    let jade = `${testRealmURL}Person/jade`;
+    let hassan = `${testRealmURL}Person/hassan`;
+
+    driver.id = hassan;
+    await waitFor(`[data-test-rendered-card="${hassan}"]`);
+    assert.strictEqual(
+      store.getReferenceCount(jade),
+      0,
+      `reference count for ${jade} is 0`,
+    );
+    assert.strictEqual(
+      store.getReferenceCount(hassan),
+      1,
+      `reference count for ${hassan} is 1`,
+    );
+
+    driver.id = jade;
+    await waitFor(`[data-test-rendered-card="${jade}"]`);
+    assert.strictEqual(
+      store.getReferenceCount(jade),
+      1,
+      `reference count for ${jade} is 1`,
+    );
+    assert.strictEqual(
+      store.getReferenceCount(hassan),
+      0,
+      `reference count for ${hassan} is 0`,
+    );
+
+    driver.showComponent = false;
+    await waitFor(`[data-test-rendered-card]`, { count: 0 });
+    assert.strictEqual(
+      store.getReferenceCount(jade),
+      0,
+      `reference count for ${jade} is 0`,
+    );
+    assert.strictEqual(
+      store.getReferenceCount(hassan),
+      0,
+      `reference count for ${hassan} is 0`,
+    );
+  });
```

Comment:
I appreciate the granularity of these tests, might be time soon to split the file up!